src/community/nsg-profile/src/test/java/org/geoserver/nsg/timeout/TimeoutTest.java

```diff
@@ -0,0 +1,150 @@
+/* (c) 2017 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.nsg.timeout;
+
+import org.geoserver.config.GeoServer;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.wfs.WFSInfo;
+import org.geoserver.wfs.v2_0.WFS20TestSupport;
+import org.geotools.data.simple.SimpleFeatureStore;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.vfny.geoserver.servlets.ServiceStrategyFactory;
+import org.w3c.dom.Document;
+
+import java.io.ByteArrayInputStream;
+import java.util.List;
+
+import static org.custommonkey.xmlunit.XMLAssert.assertXpathEvaluatesTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class TimeoutTest extends WFS20TestSupport {
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+
+        SimpleFeatureStore fs = (SimpleFeatureStore) getCatalog().getFeatureTypeByName
+                ("Fifteen").getFeatureSource(null, null);
+        // double the number of features to get a large amount of them, breaking
```

Comment:
We are actually multiplying them by 5 right ?

Reply:
Nope, each time the loop iterates, the number of items doubles, so you get 30 at the end of the first loop, 60 and the second, and so on: 30, 60, 120, 240, 480. I'll explain it better in the comment.